x-pack/solutions/observability/plugins/inventory/server/routes/entity_definition/extract_entity_index_patterns_from_definitions.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EntitySourceDefinition } from '@kbn/entityManager-plugin/server/lib/v2/types';
+
+export const extractEntityIndexPatternsFromDefinitions = (
+  entityDefinitionsSource: EntitySourceDefinition[]
+) =>
+  entityDefinitionsSource.reduce(
+    (acc, { ['type_id']: typeId }) => (
+      (acc[typeId] = [
+        ...new Set(
+          entityDefinitionsSource
+            .filter((sourceToFilter) => sourceToFilter.type_id === typeId)
+            .flatMap((filteredSource) => filteredSource.index_patterns)
+        ),
```

Comment:
can we use the accumulator instead of looping over the sources on every iteration: `uniq(concat(acc[typeId], source.index_patterns))` ?

Reply:
Thanks for the suggestion, I forgot to change that part when I extracted the logic, sorry. Fixed in 
1f03ce0